### ISSUE TITLE
Created invalid query string by use collection operators

### ISSUE THREAD

**Issue by onjson** [techniq/odata-query#27]
This filter:
```javascript
const filter = {
  Results: {
    any: {
      ResultsProperties: {
        any: {
          SomeProp: 1
        }
      }
    }
  }
}
```
Create next query string: 
`?$filter=Results/any(r:r/ResultsProperties/any(r:r/SomeProp eq 1))`
OData failed by this query with next exception:
`The range variable 'r' has already been declared.`

**Comment by techniq** [comment]
Sorry for the delay.  This is due to the naive implementation of picking a lambda variable name to set for the expression. 

https://github.com/techniq/odata-query/blob/4914b735566091770b951e77c21becffa34a460f/src/index.js#L195-L202

Since both properties start with r, we they both getting the same name.  Not sure if we couldn't track the variable names handed out and suffix with a number (`r`, `r2`, etc) or some better solution.

I'm not sure when I'll get time to be able to solve this (under some delivery deadlines as always), but I can review a PR if you'd be willing to tackle this.

**Comment by relair** [comment]
I've run into this problem myself as well, the easiest workaround is renaming properties so they start with different letter like 'Results' and 'Properties' (or 'AllResultsProperties'), but it is only possible if you have control over the API.

**Comment by techniq** [comment]
@relair @onjson You might be able to use the full `filterKey` (ie. property name) instead of just the first letter (see my comment above)  If you send a PR with the change (and updated tests), I'll review and cut a release.  Thanks

**Comment by techniq** [comment]
Fixed via #35 by @fabio-b and released as [5.6.1](https://github.com/techniq/odata-query/releases/tag/v5.6.1)
